lib/mayaUsd/ufe/UsdAttributes.cpp

```diff
@@ -471,6 +474,39 @@ static void removeConnections(const PXR_NS::UsdPrim& prim, const PXR_NS::SdfPath
                 }
             }
         }
+    };
+
+    // Remove the connections to the prims with the same parent.
+    for (const auto& node : parentPrim.GetChildren()) {
+        removeConnectionsWithPrim(node);
+    }
+
+    // Remove the connections to the parent prim.
+    removeConnectionsWithPrim(parentPrim);
+}
+
+void UsdAttributes::removeAttributesConnections(const PXR_NS::UsdPrim& prim)
+{
+    const auto primParent = prim.GetParent();
+
+    if (!primParent) {
+        return;
+    }
+
+    const PXR_NS::SdfPath kPrimPath = prim.GetPath();
+    const auto            kPrimAttrs = prim.GetAttributes();
+
+    for (const auto& attr : kPrimAttrs) {
+        const PXR_NS::SdfPath kPropertyPath = kPrimPath.AppendProperty(attr.GetName());
+        const auto            kBaseNameAndType
+            = PXR_NS::UsdShadeUtils::GetBaseNameAndType(PXR_NS::TfToken(attr.GetName()));
+
+        if (kBaseNameAndType.second == PXR_NS::UsdShadeAttributeType::Output) {
+            removeConnections(primParent, kPropertyPath);
+        }
+        if (kBaseNameAndType.second == PXR_NS::UsdShadeAttributeType::Input) {
+            removeConnections(prim, kPropertyPath);
```

Comment:
Note: this is the case for the compound, we want to delete the connections to its children. For the delete feature, this case is redundant, since we are deleting also the compound children and so all the children connections. But I think we should include this case in this function since for generic cases, not only deleting a prim, is a necessary step.